email-utils.js

```diff
@@ -0,0 +1,83 @@
+"use strict";
+
+const AppConstants = require("./app-constants");
+
+const nodemailer = require("nodemailer");
+
+// This is set later when reading SMTP credentials from the environment.
+// This exists as a variable so we can use it in the from header of emails.
+var kSMTPUsername;
+
+// The SMTP transport object. This is initialized to a nodemailer transport
+// object while reading SMTP credentials, or to a dummy function in debug mode.
+var gTransporter;
+
+var EmailUtils = {
+  init() {
+    // Allow a debug mode that will send JSON back to the client instead of sending emails.
+    if (AppConstants.DEBUG_DUMMY_SMTP) {
+      console.log("Running in dummp SMTP mode, /user/breached will send a JSON response instead of sending emails.");
+      gTransporter = {
+        sendMail(options, callback) {
+          callback(null, "dummy mode")
+        },
+      };
+      return Promise.resolve(true);
+    }
+    console.log("Attempting to get SMTP credentials from environment...");
+    kSMTPUsername = AppConstants.SMTP_USERNAME;
+    let password = AppConstants.SMTP_PASSWORD;
+    let host = AppConstants.SMTP_HOST;
+    let port = AppConstants.SMTP_PORT;
+    if (!(kSMTPUsername && password && host && port)) {
+      return Promise.reject("SMTP credentials could not be read from the environment");
+    }
+
+    // TODO: stop hardcoding this stuff.
```

Comment:
I think we can remove this comment, since we're grabbing these values from env vars now. ❤️